### ISSUE TITLE
focus-manager: prevents tabbing to elements added after dialog initial render

### ISSUE THREAD

**Issue by KaidenR** [pluralsight/design-system#1715]
## Bug

I’m having an issue with Dialog and the Focus Manager (love the idea and implementation of the focus manager btw). The issue is preventing me from tabbing to items inside my modal.
Basically FocusManager is not updating what it thinks is the last focusable element when the contents of the dialog update after the initial render. This prevents you from tabbing to anything that was dynamically added below the initial last-focusable element. Any ideas on what we can do to address this?

Here’s a [CodeSandbox](https://codesandbox.io/s/codesandboxer-example-forked-hzyuo?file=/example.tsx) that illustrates the problem.

Here’s the [actual dialog](https://ps-dev.github.io/skills-management-widgets/?path=/story/assign-manage-modals-manageassignmentsmodal--action-dropdown) having the issue (you can’t tab past the search box)

### Expected behavior

Should be able to tab to all content in dialog

### Actual behavior

Cannot tab to elements added after the dialog initial render

### Related packages

- dialog
- focus-manager

### Environment

- OS - MacOS
- Browser - Chrome

**Comment by jaketrent** [comment]
Hey Kaiden, thanks for the bug notice.  We'll take a look at it.

**Comment by KaidenR** [comment]
Any idea when you'll be able to get to this issue?

**Comment by izikandrw** [comment]
Hi @jaketrent, we upgrade to the latest version of ps-design-system-dialog (which as you know pulls in focusmanager) but we're still seeing this issue. Happy to jump on a zoom call or something to show you. Thanks!

**Comment by jaketrent** [comment]
:sad-trombone:  well, that's no good. I'll put this back on my list today to repro. Perhaps you could try to do the same. The repro codesandbox that kaiden sent was great last time. Mind doing that again with the latest ver?

**Comment by jaketrent** [comment]
I took the old sandbox and used the new version of Dialog that has the new version of FocusManager in it. Same as you, it doesn't work. I put it into our storybook, still doesn't work. Sits right next to another dynamic child story that I thought was proving it worked. I'll figure out what the diff is...

**Comment by jaketrent** [comment]
@izikandrw k, found the issue. PR is now in https://github.com/pluralsight/design-system/pull/1813
